aptos-move/aptos-gas/src/transaction/mod.rs

```diff
@@ -138,6 +165,57 @@ impl TransactionGasParameters {
         }
     }
 
+    /// New formula to charge storage fee for write set items based on fixed APT costs.
+    pub fn calculate_write_set_storage_fee<'a>(
+        &self,
+        ops: impl IntoIterator<Item = (&'a StateKey, &'a WriteOp)>,
+    ) -> Fee {
+        let excess_fee = |size: NumBytes| -> Fee {
+            match size.checked_sub(self.free_write_bytes_quota) {
+                Some(excess) => excess * self.storage_fee_per_excess_state_byte,
+                None => 0.into(),
+            }
+        };
+
+        let mut fee = Fee::zero();
+        let mut new_slots = NumSlots::zero();
+
+        for (key, op) in ops {
```

Comment:
Tried. That didn't work well.
1. `charge_write_set_gas` and `charge_storage_fee` use different contexts (`StorageGas` vs `TransactionGasParameters`), so combining them would require you to create a function that takes in both, which seems unnecessarily complicated.
2. It would also make the versioning logic more messy.